### ISSUE TITLE
Bug: no-useless-escape false alarm example

### ISSUE THREAD

**Issue by luckylooke** [eslint/eslint#16004]
### Environment

Node version: v14.19.2
npm version: 6.14.17
Local ESLint version: eslint@7.32.0 
Global ESLint version: (empty)
Operating System: MacOS 12.4

### What parser are you using?

Default (Espree)

### What did you do?

Hi, my code hits no-useless-escape rule
`450:35  error  Unnecessary escape character: \[  no-useless-escape`

but escape is needed there, otherwise it does not work

<summary>Configuration</summary>

```
{
  "extends": "eslint:recommended",
  "ignorePatterns": ["source/scripts/libraries/*"],
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "globals": {
    "$": true,
    "jquery": true,
    "PageData": true
  },
  "parserOptions": {
    "ecmaVersion": 8,
    "sourceType": "module"
  },
  "rules": {
    "no-console": "warn",
    "no-undef": "warn",
    "no-unused-vars": "warn"
  }
}
```
</details>

```js
name.replace(/[\[\]]/g, '\\$&');
```

Thank you 🙏 

### What did you expect to happen?

not ot hit no-useless-escape rule

### What actually happened?

I hit no-useless-escape rule

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
it can be rewritten to `/[[\]]/g`?

**Comment by aladdin-add** [comment]
closing, as it's working as expected.

**Comment by luckylooke** [comment]
Ok, my mistake, I probably removed both escaping chars instead of just the one.. thank you for your support 🙏
